packages/vite-plugin/src/index.ts

```diff
@@ -0,0 +1,54 @@
+import { getConfig } from '@lingui/conf';
+import { createCompiledCatalog, getCatalogs, getCatalogForFile } from '@lingui/cli/api';
+import path from 'path';
+import type { Plugin } from 'vite';
+
+const fileRegex = /\.(po)$/;
+
+type LinguiConfigOpts = {
+  cwd?: string;
+  configPath?: string;
+  skipValidation?: boolean;
+}
+
+export default function lingui(linguiConfig: LinguiConfigOpts = {}): Plugin {
+  return {
+    name: 'vite-plugin-lingui',
+
+    transform(src, id) {
+      const config = getConfig(linguiConfig);
+
+      if (fileRegex.test(id)) {
```

Comment:
We should throw error here, if the regex test fails. Similarly to how the Snowpack plugin does it.

Reply:
If we throw here, it would throw in vite for every single file which is not a .po 

This statement is analog of webpack's `rule.test`, determine which modules should be processed by this transform. 

```js
         {
           test: /\.tsx?$/,
           loader: 'babel-loader',
           exclude: /node_modules/,
         },
```